### PR TITLE
Add ability for form helper labels to be hidden

### LINKED ISSUE #545: Add ability to hide form labels in a form helper
This is a required change for the Rules UI. We need a way to hide the `<label>` text in a screenreader friendly manner so that the side-by-side inputs that make up a rule declaration can be stripped of visual noise where appropriate.

The rules UI will use standard form helpers, so I'm thinking a new `show_label` option should be supported, defaulting to `true`, which allows the label to be hidden when required by the UI.

### PR DESCRIPTION
**Pull request by Stanton**
Adds screenreader friendly `hide` class so that label is visually hidden without affecting layout.

- [x] unit tests
- [x] lexicon examples
- [x] documentation

screen shot 2017-04-28 at 12 03 38

**Example:**
```
{{
    form.text({
        'label': 'foo'
        'show-label': false
    })
}}
```

**Output:**
```
<div class="form__group">
    <label class="control__label hide">foo</label>
    <div class="controls">
        <input type="text" class="form__control" />
    </div>
</div>
```

Closes #545

### CODE REVIEW

File: docs/_form-helpers/button_group.md [hunk 1]
```diff
@@ -15,12 +15,13 @@ Group a series of related buttons together on a single line. Great for creating
...
-buttons | array  | An array of `html.button()` elements
-class   | string | Classes to be applied to the button group (not the buttons inside it)
-id      | string | A unique identifier, if required
-data-*  | string | Data attributes, eg: `'data-foo': 'bar'`
+Option     | Type   | Description
+---------- | ------ | -------------------------------------------------------------
+buttons    | array  | An array of `html.button()` elements
+class      | string | Classes to be applied to the button group (not the buttons inside it)
+id         | string | A unique identifier, if required
+show-label | bool | Control visibility of the `<label>` element without affecting layout (default: true)
```
Comment (nvasileiadis) [inline docs/_form-helpers/button_group.md]: Just a couple more spaces after the bool

File: docs/_form-helpers/password.md [hunk 2]
```diff
@@ -22,6 +22,7 @@ guidance-container | string | Element to bind guidance popover scroll behaviour
 help        | string | Additional guidance information to be displayed next to the input
 label       | string | Text for the `<label>` companion element
 required    | bool   | Visually indicates that the field must be completed
+show-label  | bool | Control visibility of the `<label>` element without affecting layout (default: true)
```
Comment (nvasileiadis) [inline docs/_form-helpers/password.md]: A few more spaces after bool

File: docs/_form-helpers/radio.md [hunk 3]
```diff
@@ -14,14 +14,15 @@ Generates a radio input field. There are various different options available to
...
+Option     | Type   | Description
+---------- | ------ | ------------------------------------------------------------
+class      | string | A space separated list of class names
+guidance   | string | Text to be displayed in a popover, adds a (?) icon after the input
 guidance-container | string | Element to bind guidance popover scroll behaviour to (default `body`)
-help     | string | Additional guidance information to be displayed next to the input
-label    | string  | Text for the `<label>` companion element
-required | bool   | Visually indicates that the field must be completed
+help       | string | Additional guidance information to be displayed next to the input
+label      | string  | Text for the `<label>` companion element
```
Comment (nvasileiadis) [inline docs/_form-helpers/radio.md]: Less space after label

File: docs/_form-helpers/textarea.md [hunk 4]
```diff
@@ -23,6 +23,7 @@ guidance-container | string | Element to bind guidance popover scroll behaviour
 help        | string | Additional guidance information to be displayed next to the input
 label       | string | Text for the `<label>` companion element
 required    | bool   | Visually indicates that the field must be completed
+show-label  | bool.  | Control visibility of the `<label>` element without affecting layout (default: true)
```
Comment (nvasileiadis) [inline docs/_form-helpers/textarea.md]: Is the '.' needed?

Comment (Stanton) [inline docs/_form-helpers/textarea.md]: It's in all the other documentation, I think it's fine

**Pull request merged**